src/basic/env-util.c

```diff
@@ -884,9 +884,13 @@ int replace_env_argv(
                 char ***ret_bad_variables) {
 
         _cleanup_strv_free_ char **n = NULL, **unset_variables = NULL, **bad_variables = NULL;
-        size_t k = 0, l = 0;
+        size_t k = 0, l;
         int r;
 
+        assert(!strv_isempty(argv));
```

Comment:
hmm, i wonder if this is the right place to insist on this? i mean, sure an empty cmdline is bogus and as we learnt event a security issue, but this should probably be checked elsewhere, for this code here it's not too relevant to check this.

i am thinking here of cases where you combine a cmdline from various parts and only execute this call on some parts of it. in such a case an empty path might well be valid, who knows?

it's only theoretical, i don't think we have any uses like that.

maybe a middle ground would be to just return a soft error here, i.e. EINVAL instead of assert()?